packages/compass-context-menu/src/context-menu-provider.tsx

```diff
@@ -0,0 +1,70 @@
+import React, {
+  useCallback,
+  useEffect,
+  useState,
+  useMemo,
+  createContext,
+} from 'react';
+import type { ContextMenuContext, MenuState } from './types';
+import { ContextMenu } from './context-menu';
+import type { EnhancedMouseEvent } from './context-menu-content';
+import { getContextMenuContent } from './context-menu-content';
+
+export const Context = createContext<ContextMenuContext | null>(null);
+
+export function ContextMenuProvider({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const [menu, setMenu] = useState<MenuState>({ isOpen: false });
+  const close = useCallback(() => setMenu({ isOpen: false }), [setMenu]);
+
+  useEffect(() => {
+    function handleContextMenu(event: MouseEvent) {
+      event.preventDefault();
+      setMenu({
+        isOpen: true,
+        children: getContextMenuContent(event as EnhancedMouseEvent).map(
+          (Content, index) => <Content key={`menu-content-${index}`} />
+        ),
+        position: {
+          // TODO: Fix handling offset while scrolling
```

Comment:
Not too sure what this was referring too, just came from the prototype

Reply:
When right click menu is active, should we just disable scrolling? This seems to be how context menus work in browser

Reply:
I'm tempted to agree and it seems right to lean into the pattern of disabling scrolling (although I'm surprised that this is the established pattern).

This is however an explicit non-goal of the scope 🤔 We could amend, if you see an easy way to do it .. I can't remember why I added it - likely just to keep things simple.